osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/services/DeviceResponseMessageService.java

```diff
@@ -325,13 +325,29 @@ private void handleMaxScheduleTimeExceeded(
       final ProtocolResponseMessage message,
       final FunctionalException maxScheduleTimeExceededException) {
 
+    final FunctionalException originalOsgpException =
+        this.getOriginalOsgpException(message, maxScheduleTimeExceededException);
+
     final ProtocolResponseMessage maxScheduleTimeExceededResponseMessage =
         ProtocolResponseMessage.newBuilder()
             .messageMetadata(message.messageMetadata())
             .result(ResponseMessageResultType.NOT_OK)
-            .osgpException(maxScheduleTimeExceededException)
+            .osgpException(originalOsgpException)
             .dataObject(message.getDataObject())
             .build();
     this.domainResponseMessageSender.send(maxScheduleTimeExceededResponseMessage);
   }
+
+  private FunctionalException getOriginalOsgpException(
+      final ProtocolResponseMessage message,
+      final FunctionalException maxScheduleTimeExceededException) {
```

Comment:
parameterName suggest that exception is of type MAX_SCHEDULE_TIME_EXCEEDED. This still has to be checked inside this method. Use more generic name

Reply:
Changed to functionalException